autopush/db.py

```diff
@@ -57,13 +57,18 @@ def get_storage_table():
 
 
 class Storage(object):
-    def __init__(self, table):
+    def __init__(self, table, metrics):
         self.table = table
+        self.metrics = metrics
 
     def fetch_notifications(self, uaid):
-        notifs = self.table.query_2(consistent=True, uaid__eq=uaid,
-                                    chid__gt=" ")
-        return list(notifs)
+        try:
+            notifs = self.table.query_2(consistent=True, uaid__eq=uaid,
+                                        chid__gt=" ")
+            return list(notifs)
+        except ProvisionedThroughputExceededException:
```

Comment:
I went through for each function and looked at what the failure handler does. And for other exceptions, its going to return a 503 and ask the user to try again, which seems like the appropriate response. The sentry logger should log the context of the exception with details about it.